keyboards/evyd13/atom47/keymaps/phsc138/keymap.c

```diff
@@ -0,0 +1,390 @@
+        case SINGLE_HOLD: register_code(KC_RCTL); break;
+    }
+}
+
+void rctl_reset(qk_tap_dance_state_t *state, void *user_data) {
+    switch(rctl_tap_state.state) {
+        case SINGLE_TAP: unregister_code(KC_RSHIFT); unregister_code(KC_RBRC); break;
+        case SINGLE_HOLD: unregister_code(KC_RCTL); break;
+    }
+    rctl_tap_state.state = 0;
+}
+
+static tap lctl_tap_state = {
+    .is_press_action = 1,
+    .state = 0
+};
+
+void lctl_finished(qk_tap_dance_state_t *state, void *user_data) {
+    lctl_tap_state.state = cur_dance(state);
+    switch(lctl_tap_state.state) {
+        case SINGLE_TAP: register_code(KC_LSFT); register_code(KC_LBRC); break;
+        case SINGLE_HOLD: register_code(KC_LCTL); break;
+    }
+}
+
+void lctl_reset(qk_tap_dance_state_t *state, void *user_data) {
+    switch(lctl_tap_state.state) {
+        case SINGLE_TAP: unregister_code(KC_LSHIFT); unregister_code(KC_LBRC); break;
+        case SINGLE_HOLD: unregister_code(KC_LCTL); break;
+    }
+    lctl_tap_state.state = 0;
+}
+
+static tap d20_tap_state = {
+    .is_press_action = 1,
+    .state = 0,
+};
+
+int d20_srand = 0;
+uint32_t timer_seed;
+
+void d20_finished(qk_tap_dance_state_t *state, void *user_data) {
+    d20_tap_state.state = cur_dance(state);
+    switch(d20_tap_state.state) {
+        case SINGLE_HOLD: {
+            if(d20_srand == 0){
+                timer_seed = timer_read32();
+                srand((unsigned int)timer_seed);
+                d20_srand = 1;
+            }
+
+            SEND_STRING("Seed: ");
+            // uint32_t has max size of 4294967296
+            char SEED_STR_SIZE = 16;
+            // Initialize seed_str
+            char seed_str[SEED_STR_SIZE];
+            for (int iter=0;iter < SEED_STR_SIZE;iter++) seed_str[iter] = 0;
+
+            int i = SEED_STR_SIZE - 2; // Leave null byte on end of string
+            while (timer_seed && i >= 0) {
+                seed_str[i] = timer_seed % 10 + '0';
+                timer_seed /= 10;
+                i -= 1;
+            }
+            i++;
+
+            // Move all characters over i spaces
+            char move = i;
+            while (i < SEED_STR_SIZE -1) {
+                seed_str[i-move] = seed_str[i];
+                seed_str[i] = 0;
+                i += 1;
+            }
+
+            send_string(seed_str);
+            break;
+            // Else allow fall through
+        }
+        case SINGLE_TAP: {
+            if(d20_srand == 0){
+                timer_seed = timer_read();
+                srand((unsigned int)timer_seed);
+                d20_srand = 1;
+            }
+
+            unsigned char roll = rand() % 20 + 1;
+            char res[3];
+            res[0] = (char)((char)(roll / 10) + '0');
+            res[1] = (char)(roll % 10 + '0');
+
+            send_string(res);
+
+            break;
+                         }
+    }
+}
+
+void d20_reset(qk_tap_dance_state_t *state, void *user_data) {
```

Comment:
```suggestion
void d20_reset(tap_dance_state_t *state, void *user_data) {
```

Reply:
Fixed this comment here: [d9e04e8](https://github.com/qmk/qmk_firmware/pull/18768/commits/d9e04e80af7200159a9307be8ed3559a94052311)